apps/challenges/views.py

```diff
@@ -1849,6 +1849,41 @@ def create_challenge_using_zip_file(request, challenge_host_team_pk):
         )
     },
 )
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def check_if_any_challenge_has_finished_submission(request, challenge_pk):
+    """
+    Checks if the given challenge has any finished submissions
+    """
+    challenge = get_challenge_model(challenge_pk)
+    challenge_phases = ChallengePhase.objects.filter(challenge=challenge)
+    for challenge_phase in challenge_phases:
+        submissions = Submission.objects.filter(
+            challenge_phase=challenge_phase,
+            status="finished"
+        )
+        if submissions.exists():
+            send_slack_approval_notification(challenge_pk)
+            return Response(True)
+    return Response(False)
+
+
+def send_slack_approval_notification(challenge_id):
+    message = f"Challenge no {challenge_id} is ready for taking submmissions. You may Approve it."
```

Comment:
"Challenge {challenge_id} has finished submissions and has requested for approval. Approve it here: <URL>".

I think it should be possible to add URLs here. Check other places where we use webhooks.

The URL will be `{HOST_URL}/api/admin/challenges/challenge/{challenge_id}`. Check the environment variable where we have host URL. This will change for local, staging and production.